### PR TITLE
light refactor and nuke spinny light

### LINKED ISSUE #18861: armed nuke should get an emergency light
## Description
hiding nuke under items is an incredibly scummy tactic
giving it a small emergency light would mitigate it

### LINKED ISSUE #18870: nuke arm AnchorEntity call debug assert
## Description
nfi why it happens

**Reproduction**
0. debug
1. spawn nuke
2. nukearm 1000 <id>
3. debug assert, nuke state not set to armed as a result

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
- in C.Sh.Light renamed Component to Components WHO DID THIS AND WHY
  if you merge this in a fork replace every `using Content.Shared.Light.Component;` with `using Content.Shared.Light.Components;`
- general .Owner refactor and what not
- move emergency light spinning into its own component, RotatingLightComponent
- nuke has a spinny double light when armed

fixes #18861, fixes #18870

**Media**
emergency lights still work

flashlights and lanterns still work

nuke lights

https://github.com/space-wizards/space-station-14/assets/39013340/3fbe5809-09cf-4870-9fe9-81cfb3abb5a0

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- tweak: Armed nuclear bombs now have warning lights. Set off a nuke and watch the light spin!!!

### CODE REVIEW

**Comment by PJB3005** [comment]
> nuke has a spinny light when armed

Can you also make it so that the nuke has different sprite states for "armed" and "about to blow"? I know it does in SS13.

**Comment by deltanedas** [comment]
i will try steal sprite then add an appearance thing for it

**Comment by dmnct** [comment]
At some point we will probably want to add other types of animated lights, like blinking, pulsating etc
Will we then create a different system and component for every kind of different light animations we could think of, (potentioally ending with something having 3-4 different LightComponents for it to both rotate and flicker and change color over time) or should we just maybe go ahead of that and call this AnimatedLightSystem and Component?

**Comment by deltanedas** [comment]
well blinking lights is part of a different light animation thing right, flashlight and flares use it

**Comment by deltanedas** [comment]
also at the very least i have currently unhardcoded the speed but i think putting the animation itself in prototypes would be annoying

File: Content.Server/Nuke/NukeSystem.cs [hunk 1]
```diff
@@ -459,6 +459,7 @@ public void ArmBomb(EntityUid uid, NukeComponent? component = null)
         _pointLight.SetEnabled(uid, true);
 
         _itemSlots.SetLock(uid, component.DiskSlot, true);
+        _transform.SetCoordinates(uid, nukeXform, nukeXform.Coordinates.SnapToGrid());
```
[the remark this answers is not in the document]
Comment (PJB3005) [reply Content.Server/Nuke/NukeSystem.cs:462]: The better fix is this:
image

I ran into this while trying out the nuke myself just now. Not your code.

**Review by Partmedia — CHANGES_REQUESTED** [review]
This superfluous namespace change from `using Content.Shared.Light.Component` to `using Content.Shared.Light.Components` will likely cause unnecessary pain for forks.

File: Content.Server/Nuke/NukeSystem.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (deltanedas) [reply Content.Server/Nuke/NukeSystem.cs:462]: using your troll then

**Comment by deltanedas** [comment]
it causes pain for upstream on its own and will forever unless its changed may as well do it now than never

**Comment by rbertoche** [comment]
how does an s causes pain?
also please stop calling changes that break public APIs refactors
it's simply a misnomer, if well named stuff is taken in consideration

**Comment by JoeHammad1844** [comment]
rotate the light source from flashlights by 90 degrees so that it lines up with the source on the sprite, but that may be out of scope

**Comment by metalgearsloth** [comment]
> This superfluous namespace change from Content.Shared.Light.Component to Content.Shared.Light.Components will likely cause unnecessary pain for forks.

Both the client and server sides suffix with s so I imagine someone forking the game right now would appreciate consistency over legacy.

**Comment by Partmedia** [comment]
As discussed at the 2023-08-19 maintainer meeting, please split up the namespace and refactoring changes into a separate PR, and add detailed information for #codebase-changes. Keep the actual content/feature changes in this PR.

**Pull request closed without merging**